rectools/models/nn/transformers/base.py

```diff
@@ -601,20 +602,34 @@ def __setstate__(self, state: tp.Dict[str, tp.Any]) -> None:
         self.__dict__.update(loaded.__dict__)
 
     @classmethod
-    def load_from_checkpoint(cls, checkpoint_path: tp.Union[str, Path]) -> tpe.Self:
-        """
-        Load model from Lightning checkpoint path.
+    def load_from_checkpoint(
+        cls,
+        checkpoint_path: tp.Union[str, Path],
+        map_location: tp.Union[str, torch.device, None] = None,
+        config_update: tp.Dict[str, tp.Any] = {},
+    ) -> tpe.Self:
+        """Load model from Lightning checkpoint path.
 
         Parameters
         ----------
         checkpoint_path: Union[str, Path]
             Path to checkpoint location.
+        map_location: Union[str, torch.device, None], default None
+            Target device to load the checkpoint (e.g., 'cpu', 'cuda:0').
+            If None, will use the device the checkpoint was saved on.
+        config_update: tp.Dict[str, tp.Any], default '{}'
+            Contains custom values for checkpoint['hyper_parameters'].
+            Config_update has to be flattened with 'dot' reducer, before passed.
 
         Returns
         -------
         Model instance.
         """
-        checkpoint = torch.load(checkpoint_path, weights_only=False)
+        checkpoint = torch.load(checkpoint_path, map_location=map_location, weights_only=False)
+        prev_config = checkpoint["hyper_parameters"]
+        prev_config_flatten = flatten(prev_config, reducer="dot")
```

Comment:
Please use our funcs for flatten and unflatten to prevent adding extra dependencies: https://github.com/MobileTeleSystems/RecTools/blob/6ffd25bf3ca8517c8bd1a7207d6b0e64c8e1912a/rectools/utils/misc.py#L199